archinstall/lib/user_interaction.py

```diff
@@ -154,19 +154,35 @@ def ask_to_configure_network():
 	# Optionally configure one network interface.
 	#while 1:
 	# {MAC: Ifname}
-	interfaces = {'ISO-CONFIG' : 'Copy ISO network configuration to installation','NetworkManager':'Use NetworkManager to control and manage your internet connection', **list_interfaces()}
+	interfaces = {
+		'ISO-CONFIG' : 'Copy ISO network configuration to installation',
+		'NetworkManager':'Use NetworkManager to control and manage your internet connection',
+		**list_interfaces()
+	}
 
-	nic = generic_select(interfaces.values(), "Select one network interface to configure (leave blank to skip): ")
+	nic = generic_select(interfaces, "Select one network interface to configure (leave blank to skip): ")
 	if nic and nic != 'Copy ISO network configuration to installation':
 		if nic == 'Use NetworkManager to control and manage your internet connection':
 			return {'nic': nic,'NetworkManager':True}
-		mode = generic_select(['DHCP (auto detect)', 'IP (static)'], f"Select which mode to configure for {nic}: ")
-		if mode == 'IP (static)':
+
+		# Current workaround:
+		# For selecting modes without entering text within brackets,
+		# printing out this part separate from options, passed in
+		# `generic_select`
+		modes = ['DHCP (auto detect)', 'IP (static)']
+		for index, mode in enumerate(modes):
+			print(f"{index}: {mode}")
+
+		mode = generic_select(['DHCP', 'IP'], f"Select which mode to configure for {nic} or leave blank for DHCP: ",
+							 options_output=False)
```

Comment:
We probably need to pass `sort=False` here if we want to print our own list, and pass another list to `generic_select` but skip output for that call. Because Generic select could change the order and then this magic "print but use another list" won't match on indexes any longer.